### PR TITLE
Add the possibility to use an async createScript hook

### PR DESCRIPTION
**Pull request by legarsjules**
Sometimes we need to fetch content (a js script) and add it into the script tag. So I added the possibility to use createScript asynchronously.

### CODE REVIEW

File: src/features/script-load.js [hunk 1]
```diff
@@ -84,5 +84,11 @@ systemJSPrototype.instantiate = function (url, firstParentUrl) {
       }
     });
     document.head.appendChild(script);
+
+    // Note: when you use a script without `src`, the onload is not triggered after
+    // the script was added to the DOM.
+    if (!script.src) {
```
Comment (joeldenning) [inline src/features/script-load.js:90]: This is a separate feature from async createScript hooks. I'm not against it, but it should be reviewed separately as it's completely unrelated to the rest of this PR. Could you move it to a separate PR? The main thing I want to check is whether new Event() works in IE11 or not.

File: src/features/script-load.js [hunk 2]
```diff
@@ -2,8 +2,8 @@
  * Script instantiation loading
  */
 import { hasDocument } from '../common.js';
-import { systemJSPrototype } from '../system-core.js';
 import { errMsg } from '../err-msg.js';
+import { systemJSPrototype } from '../system-core.js';
```
Comment (joeldenning) [inline src/features/script-load.js:6]: Please move this import statement back to where it was, so that it doesn't show a change.

File: src/features/script-load.js (hunk 1 above)
Comment (legarsjules) [reply src/features/script-load.js]: I remove it, I will create a new PR for this thanks.

File: src/features/script-load.js (hunk 2 above)
Comment (legarsjules) [reply src/features/script-load.js]: Done, sorry about that it was my vscode plugin to arrange imports

File: docs/hooks.md [hunk 3]
```diff
@@ -35,14 +35,16 @@ System.constructor.prototype.createContext = function (url) {
 };
 ```
 
-#### createScript(url) -> HTMLScriptElement
+#### createScript(url) -> Promise\<HTMLScriptElement>
```
Comment (joeldenning) [inline docs/hooks.md:38]:
```suggestion
#### createScript(url) -> HTMLScriptElement | Promise\<HTMLScriptElement>
```

File: src/features/script-load.js [hunk 4]
```diff
@@ -63,26 +63,27 @@ systemJSPrototype.instantiate = function (url, firstParentUrl) {
...
+        else {
+          var register = loader.getRegister(url);
+          // Clear any auto import registration for dynamic import scripts during load
+          if (register && register[0] === lastAutoImportDeps)
+            clearTimeout(lastAutoImportTimeout);
+          resolve(register);
+        }
+      });
+      document.head.appendChild(script);
+    })
```
Comment (joeldenning) [inline src/features/script-load.js:87]:
```suggestion
    });
```

**Review by joeldenning — APPROVED** [review]
Looks good to me

**Comment by joeldenning** [comment]
Regarding the `load` event, generally it's preferred to use the `fetch` hook rather than `createScript` when trying to execute arbitrary code. So I don't think we should simulate the load event for `<script>` elements that have textContent rather than src.

**Comment by joeldenning** [comment]
Released in https://github.com/systemjs/systemjs/releases/tag/6.11.0
